### PR TITLE
(feat) allow cors to reflect origin

### PR DESCRIPTION
**Pull request by y-nk**
Hello !

Just a small and simple PR to propose to follow up on your implementation of CORS checking. As you mentioned in the comment, it is following the express cors middleware, but missing one key feature: the ability to reflect origin when passing `true` as a setting value. You can find documentation [here](https://expressjs.com/en/resources/middleware/cors.html#configuration-options) about it.

As currently developing an api, the team i work with is really dependent on this. We need to allow local development (with localhost) to target our dev environment (not localhost) along with passing credentials (withCredentials: true) in our requests. As you may know, settings cors to `*` won't do in that case, and we were hoping to find this feature here.

Let me know if some things are missing in this PR so we can move forward.

### CODE REVIEW

**Comment by y-nk** [comment]
@icebob as mentioned in the issue, if you prefer i can go for a bigger PR. Please let me know :)

**Comment by y-nk** [comment]
@icebob sorry for keeping you waiting. I finally updated the code according to #216 latest comments and made testing.
It's worth noticing there's one breaking change here, as in default settings are *not* `"*"` anymore.

File: test/integration/index.spec.js [hunk 1]
```diff
@@ -2125,77 +2125,120 @@ describe("Test CORS", () => {
...
+				request(server)
+					.get("/test/hello")
+					.set("Origin", "http://localhost:3000")
+			)
+			.then((res) => {
 				expect(res.statusCode).toBe(200);
-				expect(res.headers["content-type"]).toBe("application/json; charset=utf-8");
 				expect(res.headers["access-control-allow-origin"]).toBe("http://localhost:3000");
-				expect(res.headers["access-control-allow-credentials"]).toBe("true");
-				expect(res.headers["access-control-expose-headers"]).toBe("X-Response-Time");
```
Comment (icebob) [inline test/integration/index.spec.js:2186]: Why do you remove checking other header properties?

File: test/integration/index.spec.js [hunk 2]
```diff
@@ -2246,72 +2315,6 @@ describe("Test CORS", () => {
 			.then(() => broker.stop()).catch(err => broker.stop().then(() => { throw err; }));
 	});
 
-	it("returns matching CORS origin wildcard with single origin", () => {
-		[broker, service, server] = setup({
```
Comment (icebob) [inline test/integration/index.spec.js:2250]: Why did you remove these test cases?

Comment (y-nk) [reply test/integration/index.spec.js:2250]: I generally deleted few of the testcases because they overlap in testing with the new ones, but if you want i can restore them. I just thought that, since other test cases are here already, this would create noise in your test suite (as in: double testing a single feature/case + slowing down the suite)

File: test/integration/index.spec.js (hunk 1 above)
Comment (y-nk) [reply test/integration/index.spec.js:2186]: I took to remove them because I think they're not what the test is testing for. In my opinion the test cases for this feature (cors policy) should be for checking that "given a cors settings and an http request as input, the proper "allow-origin" header is returned." ; so then those headers are not necessary to be included in the test.

Please let me know if you want them back :)

Comment (icebob) [reply test/integration/index.spec.js:2186]: Yeah, but testing the `access-control-allow-credentials` and `access-control-expose-headers` headers is only here. So if you remove, you should add other tests for these header values.

File: test/integration/index.spec.js (hunk 2 above)
Comment (icebob) [reply test/integration/index.spec.js:2250]: Ok, but please check that there are enough test cases to cover all if statements in the cors method. If yes, you can remove, if not, please add missing tests. Thanks
